yktrace/src/tir.rs

```diff
@@ -87,7 +87,39 @@ impl TirTrace {
                     .map(TirOp::Statement)
             );
 
-            // FIXME: Convert the block terminator to a guard if necessary.
+            // Convert the block terminator to a guard if necessary.
+            let guard = match body.blocks[user_bb_idx_usize].term {
+                Terminator::Goto(_)
+                | Terminator::Return
+                | Terminator::Drop { .. }
+                | Terminator::DropAndReplace { .. }
+                | Terminator::Call { .. }
+                | Terminator::Unimplemented(_) => None,
+                Terminator::Unreachable => panic!("Traced unreachable code"),
+                Terminator::SwitchInt {
+                    local,
+                    ref values,
+                    ref target_bbs,
+                    otherwise_bb
+                } => {
+                    // Peek at the next block in the trace to see which outgoing edge was taken and
```

Comment:
Do we know that there's always another block in the trace?

Reply:
I'm working on the assumption that there is. I don't think we can stop tracing on a `SwitchInt`. Either way, it'll crash if there isn't one, so at least it's not a silent failure.

What would you prefer?

Reply:
If that's a valid assumption, let's document it. If we're unsure, I suggest an `assert` which will at least make clear to someone that we're not entirely sure if this code is correct or not.

Reply:
I agree. Just testing the change now.